src/main/java/org/folio/rest/impl/PiecesHelper.java

```diff
@@ -26,12 +27,27 @@ public PiecesHelper(Map<String, String> okapiHeaders, Context ctx, String lang)
     // On success set id of the created entity to piece object and return it back
     return createRecordInStorage(JsonObject.mapFrom(entity), resourcesPath(PIECES)).thenApply(entity::withId);
   }
-  
+
   public CompletableFuture<Void> updatePieceRecord(Piece piece) {
-    return handlePutRequest(resourceByIdPath(PIECES, piece.getId()), JsonObject.mapFrom(piece), httpClient, ctx, okapiHeaders,
-        logger);
+    JsonObject messageToEventBus = new JsonObject();
+
+    messageToEventBus.put("receivingStatusBeforeUpdate", piece.getReceivingStatus());
```

Comment:
Quick question. Let's assume that any client uses `PUT /orders/peices/<id>` endpoint to update content of the piece record along with status. Let's assume that status in the storage is `Expected` and passed `piece` record has `Received` status. The logic here gets its status (i.e. `Received`) and stores it in the JsonObject under `receivingStatusBeforeUpdate` key. Then we send PUT request to storage and once completed (i.e. the piece's status in the storage is already `Received`) trigger an event. As far as I can see `ReceiptStatusConsistency` as the first step gets piece record from the storage and compares its status with the one specified under `receivingStatusBeforeUpdate` key. It looks to me that the statuses will be always the same and the rest of the `ReceiptStatusConsistency` logic will never be done...

Reply:
I think you are correct. If we check for inconsistency before sending a message then it should work for eg. this will be the flow:
  - get a piece by id
  - check if receivingStatus is not consistent with storage before sending message to event bus. if yes only do below
  - send message
  - rest of the steps remain the same

Let me know your thoughts?